Source/WebCore/crypto/gcrypt/GCryptRFC8032.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2,1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "GCryptRFC8032.h"
+
+namespace WebCore {
+
+bool validateEd25519KeyPair(const Vector<uint8_t>& privateKey, const Vector<uint8_t>& publicKey)
+{
+    ASSERT(privateKey.size() == 32);
+    ASSERT(publicKey.size() == 32);
```

Comment:
IMO, similar to the X25519 function, the sizes should be checked and false returned in either isn't 32.

Reply:
I had this check implemented in the 'latformCheckPairedKeys', but considering other algorithms we may support in the future (eg Curve488) which may have different keys size, it's probably better to implemented in the alg specific function, as you suggested.